src/workforce-management/Controllers/ComputerController.cs

```diff
@@ -0,0 +1,136 @@
+         * Return:
+         *     Computer list view
+         */
+        [HttpGet]
+        public IActionResult Index()
+        {
+            ComputerIndex viewModel = new ComputerIndex();
+
+            // Select all computers that are not assigned to employees
+            var unassignedComputerList = (
+                from computer in context.Computer
+                where context.Employee.All(e => e.ComputerId != computer.ComputerId)
+                select computer).ToList();
+
+            // Select all computers that are assigned to employees
+            var assignedComputerList = (
+                from computer in context.Computer
+                where unassignedComputerList.All(c => c.ComputerId != computer.ComputerId)
+                select computer).Distinct().ToList();
+
+            foreach (Computer computer in assignedComputerList)
+            {
+                var assignedEmployee =  (
+                    from employee in context.Employee
+                    where employee.ComputerId == computer.ComputerId
+                    select employee).FirstOrDefault();
+
+                viewModel.ComputerDictionary.Add(computer, assignedEmployee);
+            }
+
+            foreach (Computer computer in unassignedComputerList)
+            {
+                viewModel.ComputerDictionary.Add(computer, null);
+            }
+
+            return View(viewModel);
+        }
+
+
+        /**
+         * Purpose: Add new computer form
+         * Return:
+         *     Add computer view
+         */
+        [HttpGet]
+        public IActionResult Add()
+        {
+            ComputerAdd model = new ComputerAdd();
+            return View(model);
+        }
+
+        /**
+         * Purpose: Adds a new computer to the database and can update employee if assigned
+         * Arguments:
+         *     computerAdd - ComputerAdd view model containing form data and assigned employee ID
+         * Return:
+         *     Computer list view
+         */
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Add(ComputerAdd computerAdd)
+        {
+
+            if (ModelState.IsValid)
+            {
+                Computer newComputer = computerAdd.NewComputer;
+                Employee assignedEmployee = await (
+                    from employee in context.Employee
+                    where employee.EmployeeId == computerAdd.AssignedEmployeeId
+                    select employee).SingleOrDefaultAsync();
+
+                // Add the new computer to the database
+                context.Add(newComputer);
+                await context.SaveChangesAsync();
+
+                // If employee assigned to computer, update employee
+                if (assignedEmployee != null)
+                {
+                    // Assign the new computer to the employee
+                    assignedEmployee.Computer = newComputer;
```

Comment:
Do we need to save the entire computer instance to the employee in the db or just the computerId?